mmv1/products/netapp/volume.yaml

```diff
@@ -0,0 +1,416 @@
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      An optional description of this resource.
+  # We want to get rid of this. Please don't expose.
+  # - !ruby/object:Api::Type::Integer
+  #   name: 'snapReserve'
+  #   description: |
+  #     Snap_reserve specifies percentage of volume storage reserved for snapshot storage. Default is 0 percent. Use is deprecated.
+  - !ruby/object:Api::Type::Boolean
+    name: 'snapshotDirectory'
+    description: |
+      If enabled, a NFS volume will contain a read-only .snapshot directory which provides access to each of the volume's snapshots. Will enable "Previous Versions" support for SMB.
+  - !ruby/object:Api::Type::String
+    name: 'usedGib'
+    description: |
+      Used capacity of the volume (in GiB). This is computed periodically and it does not represent the realtime usage.
+    output: true
+  - !ruby/object:Api::Type::Enum
+    name: 'securityStyle'
+    description: |
+      Security Style of the Volume. Use UNIX to use UNIX or NFSV4 ACLs for file permissions.
+      Use NTFS to use NTFS ACLs for file permissions. Can only be set for volumes which use SMB together with NFS as protocol.
+    values:
+      - :NTFS
+      - :UNIX
+    immutable: true
+    default_from_api: true
+  - !ruby/object:Api::Type::Boolean
+    name: 'kerberosEnabled'
+    description: |
+      Flag indicating if the volume is a kerberos volume or not, export policy rules control kerberos security modes (krb5, krb5i, krb5p).
+    immutable: true
+    default_value: false
+  - !ruby/object:Api::Type::Boolean
+    name: ldapEnabled
+    description: |
+      Flag indicating if the volume is NFS LDAP enabled or not. Inherited from storage pool.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'activeDirectory'
+    description: |
+      Reports the resource name of the Active Directory policy being used. Inherited from storage pool.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'kmsConfig'
+    description: |
+      Reports the CMEK policy resurce name being used for volume encryption. Inherited from storage pool.
+    output: true
+  - !ruby/object:Api::Type::Enum
+    name: 'encryptionType'
+    description: |
+      Reports the data-at-rest encryption type of the volume. Inherited from storage pool.
+    values:
+      - :ENCRYPTION_TYPE_UNSPECIFIED
+      - :SERVICE_MANAGED
+      - :CLOUD_KMS
+    output: true
+  - !ruby/object:Api::Type::Boolean
+    name: 'hasReplication'
+    description: |
+      Indicates whether the volume is part of a volume replication relationship.
+    output: true
+  - !ruby/object:Api::Type::Array
+    name: 'restrictedActions'
+    description: |
+      List of actions that are restricted on this volume.
+    item_type: !ruby/object:Api::Type::Enum
+      name: 'restrictedActions'
+      description: |
+        Settings for volumes with SMB access.
+      values:
+        - :DELETE
```

Comment:
Do we expect new values here in the future? We may want to consider a String type instead of enum, as it allows users to use new values without needing a provider upgrade

Reply:
You making a good point on extensibility. But the API itself is an enum: https://cloud.google.com/netapp/volumes/docs/reference/rest/v1/projects.locations.volumes. How would this be handled in MM if the parameter is a string in TF, ut an enum in API?

Reply:
Many, many thanks for the clarification. I am learning every day and this is great information. I will take care with using enums going forward. Having said that: I am not aware of any current plans to add more value to this enum. It might happen in the future, but very rarely. Do you recommend to change it anyway? I am about to open a new small PR for volumes and could include this change, but unsure if I should do it, since it isn't supposed to change anytie soon and the extra validation is a nice feature too ...